app/api/services/EnrolmentsAuthService.scala

```diff
@@ -16,36 +16,34 @@
 
 package api.services
 
+import api.models.auth.UserDetails
+import api.models.errors.{ClientNotAuthorisedError, InternalError}
+import api.models.outcomes.AuthOutcome
 import config.AppConfig
-
-import javax.inject.{Inject, Singleton}
-import play.api.Logger
 import uk.gov.hmrc.auth.core.AffinityGroup.{Agent, Individual, Organisation}
 import uk.gov.hmrc.auth.core._
 import uk.gov.hmrc.auth.core.authorise.Predicate
 import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
 import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals._
 import uk.gov.hmrc.auth.core.retrieve.~
 import uk.gov.hmrc.http.HeaderCarrier
-import api.models.auth.UserDetails
-import api.models.errors.{DownstreamError, UnauthorisedError}
-import api.models.outcomes.AuthOutcome
+import utils.Logging
 
+import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class EnrolmentsAuthService @Inject() (val connector: AuthConnector, val appConfig: AppConfig) {
-
-  private val logger: Logger = Logger(this.getClass)
+class EnrolmentsAuthService @Inject() (val connector: AuthConnector, val appConfig: AppConfig) extends Logging {
```

Comment:
None of the services in api/services extend BaseService. Why?

Reply:
It's outside the scope of the PR really, but a fair question. BaseService was originally for "services that call downstream services" rather than audit/enrolments etc. Its scope has diminished somewhat, to the point where it's just a tagging trait. I'd be in favour of getting rid of it entirely, so "downstream-facing" services can just use the `RequestContextImplicits` trait.

Reply:
Or could `BaseService` extend `DownstreamResponseMappingSupport
    with Logging with RequestContextImplicits` so that the individual services didn't have to?